config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/loader/ServiceLoader.java

```diff
@@ -77,7 +79,40 @@ public void load(Bundle bundle, Element element) {
         serviceEntity.setHttpMethods(httpMethods);
         serviceEntity.setProperties(properties);
 
-        bundle.getServices().put(servicePath, serviceEntity);
+        Map<String, Service> bundleService = bundle.getServices();
+
+        if (bundleService.containsKey(serviceEntity.getPath())) {
+            String duplicatePathName = handleDuplicatePathName(bundleService, serviceEntity);
+            serviceEntity.setName(duplicatePathName.substring(duplicatePathName.lastIndexOf('/') + 1));
+            serviceEntity.setPath(duplicatePathName);
+        }
+
+        bundleService.put(serviceEntity.getPath(), serviceEntity);
+    }
+
+    /**
+     * Append a value to end of name of Service in a path if service name is found in bundle.
+     * Service Id must be different and must exist in the same folder.
+     *
+     * @param bundleService bundle containing original service
+     * @param serviceEntity duplicate service with same path and service name
+     * @return a path with a numerical value appended to the end to differentiate from original service
+     */
```

Comment:
is it possible 2 services same name same folder ??

Reply:
Only when exporting. It will append value like (2) to the end of differentiate it.

Reply:
I didn't know that gateway allowed things like this.

Reply:
The Gateway doesn't. This is really to cover the case of two services/policies who have invalid characters and can potentially become the same same string. E.g. parentfolder/service|2 and parentfolder/service<>?2 would both become parentfolder/service-2 therefore creating a conflict.
If this poses a problem I'll see what I can do

Reply:
Ideally, it should create different names. Is it possible to map different unsupported characters to different supported characters? Not for all characters, but at least for instances like these..

Reply:
We can talk more about this tomorrow in person. Mapping unsupported characters to different supported characters will mitigate the issue, however it still doesn't provide 100% coverage especially with string compression (compressing adjacent duplicate values for dashes). There will still exist combinations of strings (although rare) to create conflicts. I feel like the current solution of just replacing unknown characters with dashes, and handling duplicates by adding an enumeration on the tail will suffice. Users will also be able to easily spot these conflicting names once exported. This is an edge case, I'm hoping that this doesn't happen often.

Reply:
That sounds good. If these are rare combinations, then possibly we can ask OTK to change policy names slightly to create different names. Only reason I am stretching on this is because we have "supporting overriding policy" story coming up and these stories might conflict on some level. If they will not conflict and we can deliver clean bundle, as much as possible, I am good.